src/main/java/core/basesyntax/Calculator.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Calculator implements Calculating {
+    @Override
+    public double calculate(double num1, double num2, char operator) {
+        switch (operator) {
+            case '+': return num1 + num2;
+            case '-': return num1 - num2;
+            case '*': return num1 * num2;
+            case '^': return Math.pow(num1, num2);
+            case '/': if (num2 == 0) {
+                    throw new ArithmeticException("Illegal division by 0");
+                }
+            return num1 / num2;
+            default: throw new IllegalOperatorException("Use valid operator to calculate result");
```

Comment:
```suggestion
            default: throw new IllegalOperatorException("Invalid operator " + operator);
```

Давай будем делать сообщения исключений более информативными